searchcore/src/vespa/searchcore/proton/server/storeonlyfeedview.cpp

```diff
@@ -530,9 +530,7 @@ StoreOnlyFeedView::removeIndexedFields(SerialNum, Lid, bool, OnRemoveDoneType) {
 void
 StoreOnlyFeedView::prepareRemove(RemoveOperation &rmOp)
 {
-    const DocumentId &id = rmOp.getDocumentId();
-    const document::GlobalId &gid = id.getGlobalId();
-    documentmetastore::IStore::Result inspectRes = _metaStore.inspect(gid);
+    documentmetastore::IStore::Result inspectRes = _metaStore.inspect(rmOp.getGlobalId());
     if (_params._subDbType == SubDbType::REMOVED) {
         rmOp.setDbDocumentId(DbDocumentId(_params._subDbId, inspectRes._lid));
```

Comment:
We don't want new entries in the removed sub db if rmOp is a RemoveOperationWithGid. In that case, setDbDocumentId() should not be called.

Reply:
Fixed